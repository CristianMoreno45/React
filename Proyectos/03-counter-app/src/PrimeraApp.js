import React from 'react'
import PropTypes from 'prop-types'

const PrimeraAPP = ({ saludo, subtitulo }) => {

    return (
        // Modo 1: envolver en Div
        // <div>
        //     <h1> Hola Mundo </h1>
        //     <p>Mi primera aplicación en React</p>
        // </div>
        // Modo 2: envolver en un fragmento sin envolverlo en Div en el html final
        // <Fragment>
        //     <h1> Hola Mundo </h1>
        //     <p>Mi primera aplicación en React</p>
        // </Fragment>
        // Modo 3: envolver en un fragmento sin envolverlo en Div y sin importar Fragment de 'react'
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>


    );
}
PrimeraAPP.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraAPP.defaultProps = {
    subtitulo: 'Mi primera aplicación en React'
}

export default PrimeraAPP;