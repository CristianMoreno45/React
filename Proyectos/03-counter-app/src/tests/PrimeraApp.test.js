import React from "react";
import { render } from "@testing-library/react";
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
describe('Pruebas en <PriemraApp />', () => {
    // test('debe de mostrar el mensaje "Hola, soy Goku"', () => {
    //     const saludo = 'Hola, soy Goku';
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);
    //     expect( getByText(saludo) ).toBeInTheDocument();
    // });

    test('debe mostrar <PrimeraApp /> correctamente ', () => {
        const saludo = 'Hola, soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe mostrar el subtitulo enviado por props ', () => {
        const saludo = 'Hola, soy Goku';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);
        const textoSubtitulo = wrapper.find('p').text();
        expect(textoSubtitulo).toBe(subtitulo);
    });
});
