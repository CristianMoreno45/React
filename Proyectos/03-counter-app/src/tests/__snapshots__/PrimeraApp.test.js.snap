// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PriemraApp /> debe mostrar <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Mi primera aplicaci√≥n en React
  </p>
</Fragment>
`;
